Source/WebCore/platform/network/soup/ResourceError.h

```diff
@@ -60,6 +60,8 @@ class ResourceError : public ResourceErrorBase {
     GTlsCertificate* certificate() const { return m_certificate.get(); }
     void setCertificate(GTlsCertificate* certificate) { m_certificate = certificate; }
 
+    ErrorRecoveryMethod errorRecoveryMethod() const { return ErrorRecoveryMethod::NoRecovery; }
```

Comment:
This looks really cool. So to implement HTTPS First or HTTPS Only mode for GTK/WPE, we would need to implement this, plus add an API to set the network connection integrity policy?

How did you decide which errors should result in HTTPFallback vs. which errors should not? All errors here are surely going to be network errors, and it seems you're handling errors at IP layer, TLS layer, and even HTTP layer, so it's a little unclear to me what sorts of errors should result in fallback and which should not.

Reply:
Yes! That's correct, this will require enabling the feature via the NetworkConnectionIntegrity option. The relevant error cases are somewhat a judgement call. This patch aims for falling back on the specific platform for cases where the endpoint either doesn't support HTTPS/TLS/QUIC or that endpoint web server is probably misconfigured. This assumes that the original request was valid/correctly formed, but of course that assumption could be wrong.